frontend/file/src/main/java/com/virtuslab/gitmachete/frontend/file/highlighting/MacheteAnnotator.java

```diff
@@ -80,6 +86,28 @@ private void processMacheteGeneratedEntry(MacheteGeneratedEntry macheteEntry, An
                   .format(processedBranchName))
           .range(branch).create();
     }
+
+    // saveDocumentBeforeCheck needed in order to update the state of the .git/machete VirtualFile before actual check
+    ModalityUiUtil.invokeLaterIfNeeded(NON_MODAL, () -> saveDocumentBeforeCheck(file));
+    try {
+      val branchNamesFromFile = VfsUtil.loadText(file.getVirtualFile());
+      if (branchNamesFromFile.indexOf(processedBranchName) != branchNamesFromFile.lastIndexOf(processedBranchName)) {
+        holder.newAnnotation(HighlightSeverity.ERROR,
+            getNonHtmlString("string.GitMachete.MacheteAnnotator.branch-entry-already-defined")
+                .format(processedBranchName))
+            .range(TextRange.from(branchNamesFromFile.lastIndexOf(processedBranchName), processedBranchName.length()))
+            .create();
+      }
+    } catch (PluginException | IllegalStateException ignored) { // ignore dubious IDE checks against annotation range
```

Comment:
😯 what are the examples of such exceptions? if uncaught, do they end up in an error notification (like, a one that has a link to open an issue on GitHub etc.)?

Reply:
I encountered two types of such exceptions (which are not valid in my opinion):

1. `com.intellij.diagnostic.PluginException` coming from the `com.intellij.lang.annotation.AnnotationBuilder#range(@NotNull TextRange range)` - appears when adding a duplicate entry. It says something like:
"Range must be inside element being annotated: " + currentElementRange + "; but got: " + range;
Where `currentElementRange` is expected as valid and `range` is the range defined in our code. The cause of this exception is the fact, that there is some race condition in mentioned `#range` method. In particular - it's implementation uses a variable `private final PsiElement myCurrentElement;` which often points to an element that is **not** currently checked entry. It might point to any other entry in the machete file.

2. `IllegalStateException` - that one I did not study much. It sometimes occurs when deleting a duplicate line from the file. Anyway, it did not say anything helpful but caused an error in IDE.

Reply:
> if uncaught, do they end up in an error notification

Yes, there is a red notification saying that "IDE error occured". You can click on it and see details.

<img width="1411" alt="image" src="https://user-images.githubusercontent.com/49597713/194256088-37a10220-9bf2-436b-9aae-c9bfe3cb621a.png">